e2e/common/traits/route_test.go

```diff
@@ -0,0 +1,401 @@
+				"-t", "route.tls-key-secret=" + refKey,
+				"-t", "container.port=8443",
+			).Execute()).To(Succeed())
+			Eventually(IntegrationPodPhase(ns, integrationName), TestTimeoutMedium).Should(Equal(corev1.PodRunning))
+
+			route := Route(ns, integrationName)
+			Eventually(route, TestTimeoutMedium).ShouldNot(BeNil())
+			// must wait a little time after route is created, before an http request, 
+			// otherwise the route is unavailable and the http request will fail
+			time.Sleep(waitBeforeHttpRequest)
+			code := "TLS_Reencrypt"
+			url := fmt.Sprintf("https://%s/hello?name=%s", route().Spec.Host, code)
+			response, err := httpRequest(t, url, true)
+			assert.Nil(t, err)
+			assert.Equal(t, "Hello " + code, response)
+			Expect(Kamel("delete", "--all", "-n", ns).Execute()).Should(BeNil())
+		})
+
+		// =============================
+		// TLS Route Reencrypt With Openshift Service Serving CA
+		// =============================
+
+		Eventually(IntegrationPodPhase(ns, integrationName), TestTimeoutMedium).Should(Equal(corev1.PodRunning))
+		t.Run("Route reencrypt (with openshift service CA) https works", func(t *testing.T) {
+			// --resource references a serving certificate secret not created yet, as it is to be created
+			// when the service is created and the annotation added to it
+			// as a result eventually there is the following warning message in the console, don't worry as the pod waits for the secret to be created
+			// Warn: serving-certificate Secret not found in **** namespace, make sure to provide it before the Integration can run
+			Expect(Kamel("run", "-n", ns, "files/PlatformHttpServer.java",
+				// the --resource mounts the certificates inside secret as files in the integration pod
+				// they are used only for the pod exposing the HTTP endpoint, not the router
+				"--resource", "secret:" + servingCertificateSecret + "@/etc/ssl/" + servingCertificateSecret,
+				// quarkus platform-http uses these two properties to setup the HTTP endpoint with TLS support
+				// these certificates are used only in the pod, not the router
+				"-p", "quarkus.http.ssl.certificate.file=/etc/ssl/" + servingCertificateSecret + "/tls.crt",
+				"-p", "quarkus.http.ssl.certificate.key-file=/etc/ssl/" + servingCertificateSecret + "/tls.key",
+				"-t", "route.tls-termination=reencrypt",
+				// these certificates are used only in the router to encrypt the connection from the client to the router
+				"-t", "route.tls-certificate-secret=" + refCert,
+				"-t", "route.tls-key-secret=" + refKey,
+				"-t", "container.port=8443",
+			).Execute()).To(Succeed())
+
+			// should wait a little time before the service is ready
+			time.Sleep(2 * time.Second)
+			Eventually(Service(ns, integrationName)()).ShouldNot(BeNil())
+			annotations := make(map[string]string)
+			annotations["service.beta.openshift.io/serving-cert-secret-name"] = servingCertificateSecret
+			err = addAnnotation(ns, integrationName, annotations)
```

Comment:
The annotation should not be added to the Service directly. As it is a resource owned by the Integration, it's overwritten anytime the reconciliation of the Integration happens. One option is to define the annotation on the Integration, and have it transferred on the Service with the _owner_ trait. The problem with this approach is that the annotation will be transferred on all the owned resources. Another option would be to add an `annotations` parameter to the _service_ trait, so that extra annotations can be provided only to the Service.

Reply:
I see your point and I agree. As this is just a short lived e2e test, can we have it as it is and open a new issue to track this specific service annotation issue ?

Reply:
On one hand, these tests added-value is that they script the user actions, so they are more than just short lived e2e tests. In that case, users would not add the annotation on the Service, and we still don't have an answer. Plus, I wonder if that may actually cause the instability of the test. On the other hand, I appreciate that this PR has grown larger than anticipated, so to give us the time to find a proper solution, solve that instability issue, and have that PR merged, I think we can disable this test. WDYT?

Reply:
Agreed.